idaes/generic_models/unit_models/column_models/column.py

```diff
@@ -0,0 +1,1144 @@
+        doc="""Scheme to use when transformating domain. See Pyomo
+documentation for supported schemes,
+**default** - "BACKWARD".
+**Valid values:** {
+**"BACKWARD"** - Use a BACKWARD finite difference transformation method,
+**"FORWARD""** - Use a FORWARD finite difference transformation method,
+**"LAGRANGE-RADAU""** - use a collocation transformation method}"""))
+
+    # Create individual config blocks for vapor(gas) and liquid sides
+    CONFIG.declare("vapor_side",
+                   _PhaseCONFIG(doc="vapor side config arguments"))
+
+    CONFIG.declare("liquid_side",
+                   _PhaseCONFIG(doc="liquid side config arguments"))
+
+    # =========================================================================
+
+    def build(self):
+        """
+        Begin building model (pre-DAE transformation).
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        # Call UnitModel.build to build default attributes
+        super().build()
+
+    # =========================================================================
+        """ Set argument values for vapor and liquid sides"""
+
+        # Set flow directions for the control volume blocks
+        # Gas flows from 0 to 1, Liquid flows from 1 to 0
+        
+        # TODO: Only handling countercurrent flow for now.
+        set_direction_vapor = FlowDirection.forward
+        set_direction_liquid = FlowDirection.backward
+
+    # =========================================================================
+        """ Build Control volume 1D for vapor phase and
+            populate vapor control volume"""
+
+        self.vapor_phase = ControlVolume1DBlock(default={
+            "transformation_method": self.config.transformation_method,
+            "transformation_scheme":
+                self.config.vapor_side.transformation_scheme,
+            "finite_elements": self.config.finite_elements,
+            "collocation_points": self.config.collocation_points,
+            "dynamic": self.config.dynamic,
+            "has_holdup": self.config.has_holdup,
+            "area_definition": DistributedVars.variant,
+            "property_package": self.config.vapor_side.property_package,
+            "property_package_args":
+                self.config.vapor_side.property_package_args})
+
+        self.vapor_phase.add_geometry(
+            flow_direction=set_direction_vapor,
+            length_domain_set=self.config.length_domain_set)
+
+        self.vapor_phase.add_state_blocks(
+            information_flow=set_direction_vapor,
+            has_phase_equilibrium=False)
+
+        self.vapor_phase.add_material_balances(
+            balance_type=MaterialBalanceType.componentTotal,
+            has_phase_equilibrium=False,
+            has_mass_transfer=True)
+
+        self.vapor_phase.add_energy_balances(
+            balance_type=EnergyBalanceType.enthalpyTotal,
+            has_heat_transfer=True)
```

Comment:
The energy balances for both phases also need `has_enthalpy_transfer=True` to account for enthalpy transferred as part of the mass transfer (this is probably already lumped into the heat transfer term, but should be separated for clarity and consistency).